### PR TITLE
feat: implementation of scrollWhen feature

### LINKED ISSUE #26: Show child text twice for direction="left"
it works well in long text, but it shows child text twice for direction as left with short text
Screen Shot 2020-10-13 at 7 17 48 AM

is there any solution?

### PR DESCRIPTION
**Pull request by antontsvil**
closes #26

Adds the scrollWhen property which can be either `"always"` or `"overflow"`. When set to `"overflow"`
the content will not scroll unless it overflows the width of the container. `"always"` will scroll
regardless of container's width.

These could be considered breaking changes for 2 reasons:
* The component is no longer pure, since it needs state to keep track of `disableScroll`
* ~~The default value for `scrollWhen` is `"overflow"` - IMO a bugfix, but also a breaking change if somebody preferred the original `"always"` behavior~~

### CODE REVIEW

**Comment by cdtinney** [comment]
This looks great, thank you so much.

1) Can you change the PR to merge into `develop`? The `dist` folder will be automatically built from `master` by the CD pipeline once I merge `develop` into `master` myself.
2) Is there a downside to keeping the default as `always` to avoid the breaking change? I do agree it's a bug fix, but making a bug fix opt-in rather than breaking current behaviour is desirable.

**Comment by antontsvil** [comment]
Ok made those changes. `scrollWhen` will default to always, which is the original behavior. Let me know if you want to change the demo page or the copy in it? Wasn't sure if my wording is good there :sweat_smile:

File: README.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ export default function FooComponent() {
 | `direction`   | `"right"\|\|"left"` | `"right"`                              | Horizontal direction.                          |
 | `childMargin` | `Number`          | 10                                     | Horizontal margin between children, in pixels. |
 | `children`    | React node        | None | Children to render within the marquee. |
+| `scrollWhen`  | `"overflow"\|\|"always"` | `"always"`  | Whether you scroll always or only when the content overflows its container |
```
Comment (cdtinney) [inline README.md]: "Whether the text always scrolls, or only when the content overflows the container width." would be perfect

**Comment by cdtinney** [comment]
The files in `dist` should be removed from this PR (they are generated automatically by the build pipeline). Other than that, looks great. Thank you!

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-double-marquee",
-  "version": "1.0.6",
+  "version": "1.0.7",
```
Comment (cdtinney) [inline package.json]: Sorry for missing this -- we can keep the version, it will get bumped automatically

Comment (antontsvil) [reply package.json]: fix'd

**Comment by antontsvil** [comment]
Not sure why it's failing :thinking:

**Comment by cdtinney** [comment]
Not sure either.. looking into it.

**Comment by cdtinney** [comment]
Can you rebase on to `develop` (i.e. update it)? I merged the fix for the failing build.

**Comment by antontsvil** [comment]
Nice! Looks like we're good to go :rocket:

**Comment by antontsvil** [comment]
@cdtinney just curious when will this see release?

**Comment by cdtinney** [comment]
@antontsvil Sorry, been busy -- I'll merge tonight.

**Comment by cdtinney** [comment]
@antontsvil [v1.1.0](https://github.com/cdtinney/react-double-marquee/releases/tag/v1.1.0) is released. Thanks for the feature!

**Pull request merged**